pkg/grpc/proxy/handler.go

```diff
@@ -148,13 +169,22 @@ func (s *handler) forwardServerToClient(src grpc.ServerStream, dst grpc.ClientSt
 	ret := make(chan error, 1)
 	go func() {
 		f := &codec.Frame{}
+		bufferedSends := s.bufferedCalls
+		if len(bufferedSends) > 0 {
+			for _, msg := range s.bufferedCalls {
+				if err := dst.SendMsg(msg); err != nil {
+					ret <- err
+					break
+				}
+			}
+		}
 		for i := 0; ; i++ {
 			if err := src.RecvMsg(f); err != nil {
 				ret <- err // this can be io.EOF which is happy case
 				break
 			}
+			s.bufferedCalls = append(s.bufferedCalls, f)
```

Comment:
1. Isn't this a memory leak? we keep adding frames but never cleaning them up.
2. Shouldn't a call be added to `bufferedCalls` only if there's an error?

Reply:
We do reset `s.bufferedCalls` at the start of the call, but we could also clean it after we're done.

Regardless, this is actually how the grpc middleware does this (albeit with locking): https://github.com/grpc-ecosystem/go-grpc-middleware/blob/560829fc74fcf9a69b7ab01d484f8b8961dc734b/retry/retry.go#L158

Reply:
Lets clean it as soon as we're done, please :)

Reply:
Done!